addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/NetworkHandlerFactory.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.network;
+
+import static org.openhab.binding.network.NetworkBindingConstants.*;
+
+import java.util.Dictionary;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.network.handler.NetworkHandler;
+import org.openhab.binding.network.handler.NetworkHandlerBuilder;
+import org.osgi.service.component.ComponentContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The handler factory retrieves the binding configuration and is responsible for creating
+ * PING_DEVICE and SERVICE_DEVICE handlers.
+ *
+ * @author David Graeff
+ * @author Marc Mettke
+ */
+public class NetworkHandlerFactory extends BaseThingHandlerFactory {
+    private final Logger logger = LoggerFactory.getLogger(NetworkHandler.class);
```

Comment:
No it has some undesired behaviour with inheritance, the slf4j FAQ suggest a copy safe way but that is not yet used anywhere and pretty obscure. So just correct it 🙂